File: prophecies/apps/frontend/src/models/AlternativeValue.js
import { Model } from '@vuex-orm/core'
import ChoiceGroup from '@/models/ChoiceGroup'

export default class AlternativeValue extends Model {
  static entity = 'AlternativeValue'

  static fields () {
    return {
      id: this.attr(null),
      name: this.string(),
      value: this.string(),
      choiceGroupId: this.attr(null),
      choiceGroup: this.belongsTo(ChoiceGroup, 'choiceGroupId')
    }
  }
}

File: prophecies/core/models/alternative_value.py
from django.db import models
from prophecies.core.models.choice_group import ChoiceGroup


class AlternativeValue(models.Model):
    name = models.CharField(max_length=100)
    value = models.CharField(max_length=100, blank=True)
    choice_group = models.ForeignKey(ChoiceGroup, on_delete=models.CASCADE, related_name='alternative_values')

    def __str__(self):
        return self.name

    def save(self, *args, **kwargs):
        if not self.value:
            self.value = self.name
        super().save(*args, **kwargs)
